src/curve.rs

```diff
@@ -240,6 +243,16 @@ impl Identity for ExtendedPoint {
                        Z: FieldElement::one(),
                        T: FieldElement::zero() }
     }
+
+    fn is_identity(&self) -> bool {
+        if self.X == FieldElement::zero() &&
+           self.Y == FieldElement::one()  &&
+           self.Z == FieldElement::one()  &&
+           self.T == FieldElement::zero() {
+               return true;
```

Comment:
This code rejects, e.g., (0:2:2:0), which is the identity point.  IMO a clean way to implement this instead is to compress the point to bytes and compare with the bytes of the identity point.